dbms/src/Dictionaries/PolygonDictionary.h

```diff
@@ -0,0 +1,281 @@
+        DECLARE(Float64)
+        DECLARE(Decimal32)
+        DECLARE(Decimal64)
+        DECLARE(Decimal128)
+#undef DECLARE
+
+    void getString(const std::string & attribute_name, const Columns & key_columns, const DataTypes &, ColumnString * out) const;
+
+#define DECLARE(TYPE) \
+    void get##TYPE( \
+        const std::string & attribute_name, \
+        const Columns & key_columns, \
+        const DataTypes &, \
+        const PaddedPODArray<TYPE> & def, \
+        ResultArrayType<TYPE> & out) const;
+        DECLARE(UInt8)
+        DECLARE(UInt16)
+        DECLARE(UInt32)
+        DECLARE(UInt64)
+        DECLARE(UInt128)
+        DECLARE(Int8)
+        DECLARE(Int16)
+        DECLARE(Int32)
+        DECLARE(Int64)
+        DECLARE(Float32)
+        DECLARE(Float64)
+        DECLARE(Decimal32)
+        DECLARE(Decimal64)
+        DECLARE(Decimal128)
+#undef DECLARE
+
+    void getString(
+            const std::string & attribute_name,
+            const Columns & key_columns,
+            const DataTypes &,
+            const ColumnString * const def,
+            ColumnString * const out) const;
+
+#define DECLARE(TYPE) \
+    void get##TYPE( \
+        const std::string & attribute_name, \
+        const Columns & key_columns, \
+        const DataTypes &, \
+        const TYPE def, \
+        ResultArrayType<TYPE> & out) const;
+        DECLARE(UInt8)
+        DECLARE(UInt16)
+        DECLARE(UInt32)
+        DECLARE(UInt64)
+        DECLARE(UInt128)
+        DECLARE(Int8)
+        DECLARE(Int16)
+        DECLARE(Int32)
+        DECLARE(Int64)
+        DECLARE(Float32)
+        DECLARE(Float64)
+        DECLARE(Decimal32)
+        DECLARE(Decimal64)
+        DECLARE(Decimal128)
+#undef DECLARE
+
+    void getString(
+            const std::string & attribute_name,
+            const Columns & key_columns,
+            const DataTypes & key_types,
+            const String & def,
+            ColumnString * const out) const;
+
+    /** Checks whether or not a point can be found in one of the polygons in the dictionary.
+     *  The check is performed for multiple points represented by columns of their x and y coordinates.
+     *  The boolean result is written to out.
+     */
+    // TODO: Refactor the whole dictionary design to perform stronger checks, i.e. make this an override.
+    void has(const Columns & key_columns, const DataTypes & key_types, PaddedPODArray<UInt8> & out) const;
+
+protected:
+    /** The number of dimensions used. Change with great caution, some extra work will be required. */
+    static constexpr size_t DIM = 2;
+    /** A point in Euclidean coordinates. */
+    using Point = bg::model::point<Float64, DIM, bg::cs::cartesian>;
+    /** A polygon in boost is a an outer ring of points with zero or more cut out inner rings. */
+    using Polygon = bg::model::polygon<Point>;
+    /** A multi_polygon in boost is a collection of polygons. */
+    using MultiPolygon = bg::model::multi_polygon<Polygon>;
```

Comment:
It seems that we don't need `multi_polygon`, cause dictionary is an already set of polygons.

Reply:
Switched to storing polygons instead of multi-polygons.